framework/framework_test.go

```diff
@@ -4,114 +4,497 @@ import (
 	"context"
 	"reflect"
 	"testing"
-)
-
-// Test_Framework_ProcessCreate_NoResource ensures there is an error thrown when
-// executing ProcessCreate without having any resources provided.
-func Test_Framework_ProcessCreate_NoResource(t *testing.T) {
-	err := ProcessCreate(context.TODO(), nil, nil)
-	if !IsExecutionFailed(err) {
-		t.Fatal("expected", true, "got", false)
-	}
-}
-
-// Test_Framework_ProcessDelete_NoResource ensures there is an error thrown when
-// executing ProcessDelete without having any resources provided.
-func Test_Framework_ProcessDelete_NoResource(t *testing.T) {
-	err := ProcessDelete(context.TODO(), nil, nil)
-	if !IsExecutionFailed(err) {
-		t.Fatal("expected", true, "got", false)
-	}
-}
```

Comment:
I rewrote the tests to have them all table driven. I also added new tests for the cancelation.